sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -165,7 +165,10 @@ statement
     | CREATE (OR REPLACE)? TEMPORARY? FUNCTION (IF NOT EXISTS)?
         identifierReference AS className=stringLit
         (USING resource (COMMA resource)*)?                            #createFunction
-    | DROP TEMPORARY? FUNCTION (IF EXISTS)? identifierReference      #dropFunction
+    | DROP TEMPORARY? FUNCTION (IF EXISTS)? identifierReference        #dropFunction
+    | DECLARE (OR REPLACE)? VARIABLE?
+        multipartIdentifier dataType? variableDefaultExpression?       #createVariable
```

Comment:
why can data type be optional?

Reply:
We can derive it from the default. So you can do:
DECLARE var = 'hello';

Reply:
ah I see, but at least one of them should be specified.

Reply:
Yes

Reply:
Ugh, I didn't block that and produce a void. Maybe a bad idea, will fix

Reply:
Fixed